src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyItemSourceInfo.cs

```diff
@@ -70,5 +70,20 @@ public override int GetHashCode()
         {
             return info == null ? null : new DependencyItemSourceInfo(DependencyItemSourceType.File, info);
         }
+
+        public static bool operator ==(DependencyItemSourceInfo a, DependencyItemSourceInfo b)
+        {
+            if (ReferenceEquals(a, b))
+            {
+                return true;
+            }
+
+            return (object)a != null && a.Equals(b);
```

Comment:
i'd like to check whether a is null, however i couldn't use `!=` directly because it is being overwritten, which would otherwise lead to dead loop

---
In reply to: [104119610](https://github.com/dotnet/docfx/pull/1383#discussion_r104119610) [](ancestors = 104119610)